libhb/qsv_common.c

```diff
@@ -4507,11 +4484,15 @@ static int hb_qsv_ffmpeg_set_options(hb_job_t *job, AVDictionary** dict)
     int err;
     AVDictionary* out_dict = *dict;
 
-    if (job->qsv.ctx && job->qsv.ctx->qsv_device)
+    if (job->qsv.ctx && job->qsv.ctx->dx_index > 0)
```

Comment:
In this case -1 is invalid index here and decoder initialized with default 0 number which is non Intel device and fails